lib/iris/mesh/components.py

```diff
@@ -2808,6 +2759,31 @@ def location(self):
     def axis(self):
         return self._metadata_manager.axis
 
+    @property
+    def points(self):
+        """The coordinate points values as a NumPy array."""
+        if self.timestamp < self.mesh.timestamp or self.timestamp is None:
+            self._values, _, _ = self._load_points_and_bounds()
```

Comment:
This line updates `self.timestamp`. This means that accessing `points` first will prevent `bounds` from being updated, and the same in reverse.

It would be safer to update both `points` and `bounds` when EITHER is accessed.